view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java

```diff
@@ -356,7 +370,64 @@ public static TableConfig buildPinotTableConfig(
     return tableConfig;
   }
 
-  private static IngestionConfig getTableIngestionConfig(@Nullable PinotTableSpec tableSpec) {
+  private static RoutingConfig toRoutingConfig(Config routingConfig) {
+    if (routingConfig == null) {
+      return null;
+    }
+    List<String> segmentPrunerTypes = null;
+    if (routingConfig.hasPath("segmentPrunerTypes")) {
+      segmentPrunerTypes = routingConfig.getStringList("segmentPrunerTypes");
+    }
+    String instanceSelectorType = getOptionalString(routingConfig, "instanceSelectorType", null);
+    if (segmentPrunerTypes != null || instanceSelectorType != null) {
+      return new RoutingConfig(null, segmentPrunerTypes, instanceSelectorType);
+    }
+    return null;
+  }
+
+  private static SegmentPartitionConfig toSegmentPartitionConfig(Config segmentPartitionConfig) {
+    if (segmentPartitionConfig == null) {
+      return null;
+    }
+    Config columnPartitionMap = segmentPartitionConfig.getConfig("columnPartitionMap");
+    Map<String, ColumnPartitionConfig> tableColumnPartitionMap = Maps.newHashMap();
+    for (Entry<String, ConfigValue> entry : columnPartitionMap.entrySet()) {
+      String columnName = ConfigUtil.splitPath(entry.getKey()).get(0);
```

Comment:
Do we have an example of this config? Just curious, why do we have to extract columnName via splitPath

Reply:
splitting is needed to extract the dynamic keys. this map (`tableColumnPartitionMap `) is expected while building table config

Reply:
example config is part of the PR itself. check the test config added in PR.

Reply:
This is the one, right?

```
segmentPartitionConfig = {
    columnPartitionMap = {
      tenant_id = {
        functionName = "Modulo",
        numPartitions = 4
      }
    }
  }
```

Reply:
yes